### ISSUE TITLE
Window dragged by titlebar from a "snapped position" is offset relatively to the mouse

### ISSUE THREAD

**Issue by T3sT3ro** [domferr/tilingshell#60]
**Describe the bug**
When I try to drag a window by titlebar from the maximized view or from a quarter tiling, the mouse is offset to the right relatively to the window it's dragging.

**To Reproduce**
1. Open the extension settings window
2. Resize it to around 1/3 width and length of the screen
3. maximize the window by dragging it to the top of the screen
4. now, grab the window by the title bar **on the right side**, right next to the close/maximize/minimize icons

**Screenshots**
https://github.com/domferr/tilingshell/assets/5300963/1115008a-0b52-4e68-980d-73c778780d92

**Information (please complete the following):**
 - Tiling Shell version: v10.0
 - GNOME version: GNOME Shell 42.9

**Comment by domferr** [comment]
Hi, thank you for reporting this bug!

It does happen all the time right? Doesn't matter if you grab on the right or on the left?

**Comment by domferr** [comment]
Uh, since I haven't GNOME 42.9 and I can't reproduce the problem, I have another question: does it happen even if the window IS NOT maximized? Thank you, let's fix this!

**Comment by T3sT3ro** [comment]
> Hi, thank you for reporting this bug!
> 
> It does happen all the time right? Doesn't matter if you grab on the right or on the left?

Yes, it happens all the time, although the window is offset to the left, so grabbing by the right side makes it more visible. Grabbing it by the left side could make it seem like it's working, but it may as well just happen to be offset just-right

**Comment by T3sT3ro** [comment]
I have some intuition about why this might be happening:
The window that is in the snapped area has certain width, and when it's dragged by the titlebar it's reverting to it's original size. So when the window was initially smaller, and got stretched by snapping to the bigger area (maximized or to a corner area), the mouse "pivot/drag point" is calculated from the area width, and that is then used to offset the now smaller window. I tried eyeball testing it and it seems to be that: when the window is dragged by the title bar from it's snapped position it snaps to the left edge of it's layout area instead, but it should snap to the mouse instead

**Comment by domferr** [comment]
Thank you for the details, you're awesome!
I tried to run a VM, but I didn't get it working for some reasons.

Thanks to the details you shared, I tried to implement a bug fix (but without testing it on GNOME 42). I will try again later this day to run a VM with GNOME 42 (probably Fedora 36).

Meanwhile, if you can test it, I leave here the update! If you do, please let me know if the bug is fixed. :crossed_fingers: 

v10.1 for GNOME 42, 43 and 44: [GNOME.42-44.tilingshell@ferrarodomenico.com.zip](https://github.com/user-attachments/files/16033248/GNOME.42-44.tilingshell%40ferrarodomenico.com.zip)

**Comment by domferr** [comment]
Ok, it is NOT fixed ahahha
I'm able to assign the right offset to the window when it is grabbed, but then for some reasons unknown to me, GNOME 42 reassigns again the old position (which has a bad offset). Indeed in this video, the window gets positioned in the right place for a moment and then its position is set back with a wrong offset. The first time, the position is assigned by Tiling Shell, while the second time by GNOME.

[Screencast from 2024-06-28 18-25-19.webm](https://github.com/domferr/tilingshell/assets/14203981/44a3a933-bad5-46fb-b10a-51bf109dd953)

I'm sorry to say that this needs more time to be fixed. I'm happy that, thanks to you, I can reproduce the problem! Hope to have news soon.

Meanwhile, you can disable from the preferences the feature "_Restore window size_", that puts the window back to the old size, so you can enjoy Tiling Shell without this bug!

![image](https://github.com/domferr/tilingshell/assets/14203981/06e49356-dce8-47fa-8558-1edcb47324c6)

**Comment by domferr** [comment]
@T3sT3ro I made a bug fix, which solves the problem on my VM with GNOME 42. I leave here the files with the bugfix, so you can try it in advance. If you do, please let me know if it solves the problem!

[GNOME.42-44.tilingshell@ferrarodomenico.com.zip](https://github.com/user-attachments/files/16037247/GNOME.42-44.tilingshell%40ferrarodomenico.com.zip)

**Comment by T3sT3ro** [comment]
Hello, I tested the zip you sent and it looks like the initial issue is solved!

I noticed another, maybe related bug, however: if you click and move the mouse fast enough, the window can still be offset for some reason. Almost as if the window position was calculated on a different frame than the drag start. The attached video shows this case in the third attempt (First two attempts worked OK)

https://github.com/domferr/tilingshell/assets/5300963/2c607a14-fb80-4110-9a7f-492027357d5c
